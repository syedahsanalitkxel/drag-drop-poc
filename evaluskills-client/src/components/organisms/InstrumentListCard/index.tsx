import React from 'react';

import InstrumentCard from '../../molecules/InstrumentDetailCard';
import { ClientInstruments } from '../../../interfaces/Instruments';

interface ListCardProps {
  listData: ClientInstruments[];
  titleKey: string;
  view?: (id: string) => void;
}
const pStyle = {
  width: '50%',
};

const InstrumentListCard: React.FunctionComponent<ListCardProps> = ({
  titleKey,
  listData,
  view,
}) => {
  // TODO: Add checkbox support
  // TODO: Add support remove action handlers and replace them with CheckBox
  const actionHandler = (assessmentId: string) => (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    if (event.currentTarget.name === 'view') {
      if (view) {
        view(assessmentId);
      }
    }
  };

  const renderContent = (
    title: string,
    noOfAssessmentItems: string,
    noOfEvaluations: string,
    noOfParticipants: string,
    id: string
  ) => (
    <React.Fragment>
      <div className="d-inline-flex m-r-15 text-center">
        <span className="font-size-12">No of Assessment Items:</span>
        <span className="font-size-12 font-weight-bold">{noOfAssessmentItems}</span>
      </div>
      <div className="d-inline-flex m-r-15 text-center">
        <span className="font-size-12">No of Participants:</span>
        <span className="font-size-12 font-weight-bold">&nbsp;{noOfParticipants}</span>
      </div>
      <div className="d-inline-flex m-r-15 text-center">
        <span className="font-size-12">No of Evaluations:</span>
        <span className="font-size-12 font-weight-bold">&nbsp;{noOfEvaluations}</span>
      </div>
    </React.Fragment>
  );

  const renderProgressBar = (id: string) => (
    <React.Fragment>
      <div className="progress">
        <div style={pStyle} className="progress-bar">
          50%
        </div>
      </div>
      <span>1 Evaluations Received</span>
    </React.Fragment>
  );

  const renderActions = (id: string) => {
    return (
      <React.Fragment>
        {view && (
          <button
            id={id}
            name="view"
            type="button"
            onClick={actionHandler(id)}
            className="btn btn-default"
          >
            Detail
          </button>
        )}
      </React.Fragment>
    );
  };

  function renderAllCards(item: ClientInstruments) {
    if (item) {
      const content = renderContent(
        item.title,
        item.noOfAssessmentItems,
        item.noOfEvaluations,
        item.noOfParticipants,
        item.id
      );

      const actions = renderActions(item.id);
      const bar = renderProgressBar(item.id);

      return (
        <InstrumentCard key={item.id} header={item.title}>
          {{ content, actions, bar }}
        </InstrumentCard>
      );
    }
  }

  return <React.Fragment>{listData && listData.map(renderAllCards)}</React.Fragment>;
};

export default InstrumentListCard;
