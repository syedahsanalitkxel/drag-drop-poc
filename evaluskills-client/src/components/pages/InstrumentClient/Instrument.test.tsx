import { configure, mount, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Instrument from './index';
import InstrumentListCard from '../../organisms/InstrumentListCard/index';

configure({ adapter: new Adapter() });
describe('Instrument', () => {
  const instruments = [
    {
      id: '1',
      noOfAssessmentItems: '25',
      noOfEvaluations: '28',
      noOfParticipants: '30',
      title: '360째 Leadership Instrument',
    },
    {
      id: '2',
      noOfAssessmentItems: '25',
      noOfEvaluations: '28',
      noOfParticipants: '30',
      title: '210째 Leadership Instrument',
    },
  ];

  const mockViewFunction = jest.fn();
  const filterMockFunction = jest.fn();

  const instrumentProps = {
    listData: instruments,
    view: mockViewFunction,
    filter: filterMockFunction,
  };

  const client = shallow(<Instrument />);

  it('render successfully', () => {
    expect(client).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Instrument />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without crashing with props', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Instrument {...instrumentProps} />
      </Router>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe('InstrumentListCard', () => {
    const Key = 'title';
    const Data = [
      {
        id: '1',
        noOfAssessmentItems: '25',
        noOfEvaluations: '28',
        noOfParticipants: '30',
        title: '360째 Leadership Instrument',
      },
      {
        id: '2',
        noOfAssessmentItems: '25',
        noOfEvaluations: '28',
        noOfParticipants: '30',
        title: '210째 Leadership Instrument',
      },
    ];

    const viewMockFunction = jest.fn();

    const props = { titleKey: Key, listData: Data, view: viewMockFunction };

    it('Instrument List Card to be defined', () => {
      const wrapper = shallow(<Instrument />);
      expect(wrapper.find('InstrumentListCard')).toBeDefined();
    });
    it('Instrument List Card Render without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Router>
          <InstrumentListCard />
        </Router>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Instrument List Card Render without props', () => {
      const div = document.createElement('div');
      ReactDOM.render(
        <Router>
          <InstrumentListCard {...props} />
        </Router>,
        div
      );
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Progress bar to be defined', () => {
      const wrapper = shallow(<InstrumentListCard />);
      expect(wrapper.find('.progress')).toBeDefined();
    });
  });
});
