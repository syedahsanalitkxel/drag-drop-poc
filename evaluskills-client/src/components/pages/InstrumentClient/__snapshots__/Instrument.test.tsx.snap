// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instrument render successfully 1`] = `
<DashboardTemplate>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <PageHeader
        filterAction={[Function]}
        title="Instrument"
      />
      <PageBody
        card={false}
      >
        <InstrumentListCard
          titleKey="title"
        />
        <Pager />
      </PageBody>
    </div>
  </div>
  <ESModal
    primaryAction={[Function]}
    primaryText="Save Changes"
    secondaryAction="reset"
    secondaryText="Close"
    title="Filters"
    toggle={[Function]}
    visible={false}
  >
    <InstrumentFilters />
  </ESModal>
</DashboardTemplate>
`;
